hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClusterListenerThread.java

```diff
@@ -250,22 +250,22 @@ private ClientConnection connectToOne() throws Exception {
 
         int attempt = 0;
         Throwable lastError = null;
-        Set<Address> triedAddresses = new HashSet<Address>();
+        Set<InetSocketAddress> triedAddresses = new HashSet<InetSocketAddress>();
         while (true) {
             final long nextTry = Clock.currentTimeMillis() + connectionAttemptPeriod;
             final Collection<InetSocketAddress> socketAddresses = getSocketAddresses();
             for (InetSocketAddress isa : socketAddresses) {
-                Address address = new Address(isa);
-                triedAddresses.add(address);
-                LOGGER.finest("Trying to connect to " + address);
```

Comment:
Please make use of 'isFinestEnabled'. Even though it isn't code that very frequently runs, I think it is best to do.